src/Discord.Net.Commands/Readers/TimeSpanTypeReader.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using System.Globalization;
+using System.Threading.Tasks;
+
+namespace Discord.Commands
+{
+    internal class TimeSpanTypeReader : TypeReader
+    {
+        private static readonly string[] _formats = new[]
+        {
+            "%d'd'%h'h'%m'm'%s's'", //4d3h2m1s
+            "%d'd'%h'h'%m'm'",      //4d3h2m
+            "%d'd'%h'h'%s's'",      //4d3h  1s
+            "%d'd'%h'h'",           //4d3h
+            "%d'd'%m'm'%s's'",      //4d  2m1s
+            "%d'd'%m'm'",           //4d  2m
+            "%d'd'%s's'",           //4d    1s
+            "%d'd'",                //4d
+            "%h'h'%m'm'%s's'",      //  3h2m1s
+            "%h'h'%m'm'",           //  3h2m
+            "%h'h'%s's'",           //  3h  1s
+            "%h'h'",                //  3h
+            "%m'm'%s's'",           //    2m1s
+            "%m'm'",                //    2m
+            "%s's'",                //      1s
+        };
+
+        public override Task<TypeReaderResult> ReadAsync(ICommandContext context, string input, IServiceProvider services)
+        {
+            return (TimeSpan.TryParseExact(input, _formats, CultureInfo.InvariantCulture, out var timeSpan))
```

Comment:
What about `input.ToLower()` so input is case insensitive?

Reply:
While I would admit, as a developer, weeks seems a bit excessive, I see it used in various bots and actually by the consumers of the bots. I would consider it worth adding.

Reply:
@Chris-Johnston Will do. I *was* hoping to cache the `InvariantCulture` so the system doesn't have to excessively load it (especially because the backing field in NetFX is `volitile`), but NS1.1 doesn't support `.ToLower(CultureInfo)`. Oh well....

@khionu Sadly, [there is no format specifier for weeks](https://docs.microsoft.com/en-us/dotnet/standard/base-types/custom-timespan-format-strings), so this is as good as it's gonna get.

Reply:
Ahh... oh well. Could always manually parse, but that I think that might be considered a bit much for this lib